openssl/src/x509/mod.rs

```diff
@@ -1261,6 +1261,27 @@ impl X509NameRef {
         to_der,
         ffi::i2d_X509_NAME
     }
+
+    /// Returns a digest of the DER representation of this 'X509Name'.
+    #[corresponds(X509_NAME_digest)]
+    pub fn digest(&self, hash_type: MessageDigest) -> Result<DigestBytes, ErrorStack> {
```

Comment:
A macro may be needed to generate all digest method as described here https://www.openssl.org/docs/man1.1.1/man3/X509_digest.html.